module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -690,12 +690,76 @@ public function getMyTransactions($patron)
                 'id' => $this->getBibId($trans->item->instanceId),
                 'item_id' => $trans->item->id,
                 'barcode' => $trans->item->barcode,
+                'renew' => $trans->renewalCount ?? 0,
+                'renewable' => true,
                 'title' => $trans->item->title,
             ];
         }
         return $transactions;
     }
 
+    /**
+     * Get FOLIO loan IDs for use in renewMyItems.
+     *
+     * @param array $transaction An single transaction
+     * array from getMyTransactions
+     *
+     * @return string The FOLIO loan ID for this loan
+     */
+    public function getRenewDetails($transaction)
+    {
+        return $transaction['item_id'];
+    }
+
+    /**
+     * Attempt to renew a list of items for a given patron.
+     *
+     * @param array $renewDetails An associative array with
+     * patron and details
+     *
+     * @return array $renewResult result of attempt to renew loans
+     */
+    public function renewMyItems($renewDetails)
+    {
+        $renewalResults = [];
+
+        foreach ($renewDetails['details'] as $loanId) {
+            $renewal = [];
+            $requestbody = [
+                'itemId' => $loanId,
+                'userId' => $renewDetails['patron']['id']
+                ];
+            $response = $this->makeRequest(
+                'POST', '/circulation/renew-by-id', json_encode($requestbody)
+            );
+            if ($response->isSuccess()) {
+                $json = json_decode($response->getBody());
+                $rawDueDate = date_create($json->dueDate);
+                $renewal = [
+                    'success' => true,
+                    'new_date' =>  date_format($rawDueDate, "j M Y"),
```

Comment:
Rather than using date_create / date_format here, it is probably preferable to use the VuFind date converter, which formats dates based on user preferences in config.ini. It looks like the FOLIO driver is currently being injected with one (as `$this->dateConverter`) but then never using it. You can see how this is used in the Voyager driver here: https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/ILS/Driver/VoyagerRestful.php#L1378 -- note that the object property is named `$this->dateFormat` there, but it's the same thing.